.github/workflows/ci.yml

```diff
@@ -68,11 +69,17 @@ jobs:
         run: sbt +test
 
       - uses: actions/upload-artifact@v3
-        if: ${{ startsWith(matrix.os, 'macos') }}
+        if: matrix.os == 'macos-latest' && matrix.architecture == 'x86_64'
         with:
-          name: macos
+          name: macos-x86_64
           path: native/target/native/x86_64-darwin/bin
 
+      - uses: actions/upload-artifact@v3
+        if: matrix.os == 'flyci-macos-large-latest-m1' && matrix.architecture == 'arm64'
+        with:
+          name: macos-arm64
+          path: native/target/native/arm64-darwin/bin
```

Comment:
Thanks for this PR!

We also need modifying the publish section: https://github.com/PDAL/java/blob/main/.github/workflows/ci.yml#L132-L135 

'upload' uploads built artifacts and then we 'download' them to pack into the result package.

Reply:
I'm not really practise, should we add
```
 - uses: actions/download-artifact@v3
        with:
          name: macos-ARM
          path: native/target/native/arm64-darwin/bin
```
          
 Something like that after L 135?